Switch to faustwasm and reorganize

ijc8 opened the pull request:
Builds on #3. Instead of checking in copies of `faustwasm` and `faust-ui` into this repository, this PR creates `package.json` and lists them (along with CodeMirror 5, which is already copied on `master`) as npm dependencies, with the aim of decluttering the repo and simplifying maintenance.

WIP. Remaining tasks:
- [x] Convert remaining scripts into ES modules so `vite build` works.
- [x] Configure GitHub actions so that this will continue to work as a GitHub pages site.
- [x] Finish testing.
- [x] Update README with setup & build instructions.

Questions:
- What's the status of the faustlive-wasm stuff? ~The live version at https://fausteditor.grame.fr/faustlive-wasm.html seems to be broken.~
- What's the status of `_f4u$t`, and how does it relate to the newer faust-ui?

Fr0stbyteR committed:
use faustwasm as compiler

ijc8 committed:
Create package.json

ijc8 committed:
Add dependencies to package.json, remove copies

ijc8 committed:
Update `faustwasm-live.js`

ijc8 committed:
Minor fixes and reorganization

ijc8 committed:
Remove copy of CodeMirror from repository

and start changing scripts to modules

ijc8 committed:
Remove base64.js in favor of builtin `atob()`

base64.js seems to be third-party; can't determine its origin.

ijc8 committed:
Convert `exportUI` into a module

and cleanup `faustlive` exports

ijc8 committed:
Remove copy of `qrcode.js`, module-ify `ExportLib`

ijc8 committed:
Convert `runfaust` and `compilefaust` to modules

ijc8 committed:
Remove copy of `tippy.js` from repository

ijc8 committed:
Convert `localStorage.js` into a module

sletz commented:
- https://fausteditor.grame.fr/faustlive-wasm.html is fixed

- _f4u$t is related to the old SVG based GUI (developed by Mike Solomon in 2013). faust_ui is Shihong new GUI model, developed for Faust IDE. We can discuss if moving from f4u$t to faust_ui  can be done in fausteditor also, so that to only use and maintain one GUI model for all projects.

ijc8 committed:
Convert jsscripts to module, get build working

ijc8 committed:
Remove copy of font-awesome from repository

ijc8 committed:
Export `expandDSP` for block diagram display

ijc8 committed:
Bring back `docs.css` and reorganize

ijc8 committed:
Remove unused font and CSS rule

overriden by `docs.css`

sletz commented:
I understand the package has to be built using npm right ? can you possibly complete the README to explain how to build ? Thanks.

ijc8 committed:
Revert "Correct links."

This reverts commit 5d8368fdad8637cd7f898cc4793d6c7158a995d3.

ijc8 committed:
Export functions for remaining buttons

ijc8 committed:
Update README with build instructions

ijc8 committed:
_f4u$t fixes

ijc8 committed:
Fix up & setup build for faustlive-wasm.html

ijc8 committed:
Export `isPoly`

sletz commented:
Testing with `npm run dev`:
- configure running mode, polyphonic mode is working
- but "Save DSP control parameters state" and "Save DSP source" are not woking anymore, seems an issue with local storage ?

And `npm run build` produces a dist folder with missing CSS ? (ugly display)

ijc8 committed:
More exports

ijc8 committed:
Comment out formerly-broken CSS rules

These are inoperative on `master` because they lack spaces around the
'-' in `calc()`.

ijc8 committed:
Switch from _f4u$t to faust-ui

ijc8 committed:
Remove _f4u$t, jQuery, and jQuery SVG

ijc8 committed:
Add more detail to README

ijc8 committed:
Make tooltips more consistent

ijc8 committed:
Add GitHub Actions workflow to build with Vite

ijc8 committed:
Change `base` in Vite config

sletz commented:
BTW: I guess some files like mixer32.wasm and  libfaust-wasm.data (part of faustwasm subproject AFAICS) are not needed in the repository anymore right ?

ijc8 committed:
Switch dependency to `@grame/faustwasm`

ijc8 committed:
Remove copy of binary (`mixer32.wasm`)

ijc8 committed:
Avoid minifying identifiers for faustwasm

ijc8 committed:
Remove copy of `libfaust-wasm.data`

ijc8 committed:
Update faustwasm, remove minifier workaround
